From: Jean Delvare <jdelvare@suse.de>
Subject: idr: Print a stack dump after ida_remove warning
Patch-mainline: 3.11
Git-commit: dd04b452f532ca100f7c557295ffcbc049c77171
References: http://lists.opensuse.org/opensuse-kernel/2013-04/msg00102.html

We print a stack dump after idr_remove warning. This is useful to find
the faulty piece of code. Let's do the same for ida_remove, as it
would be equally useful there.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
Cc: Tejun Heo <tj@kernel.org>
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: Takashi Iwai <tiwai@suse.de>
---
 lib/idr.c |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/idr.c
+++ b/lib/idr.c
@@ -1039,6 +1039,7 @@ void ida_remove(struct ida *ida, int id)
 static void idr_remove_warning(int id)
 {
	WARN(1, "idr_remove called for id=%d which is not allocated.\n", id);
+	dump_stack();
 }
 
 static void sub_remove(struct idr *idp, int shift, int id)
